core/src/main/scala/cats/syntax/list.scala

```diff
@@ -1,12 +1,71 @@
 package cats
 package syntax
 
-import cats.data.NonEmptyList
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.{Ior, IorNel, NonEmptyList, ValidatedNel}
 
 trait ListSyntax {
   implicit def catsSyntaxList[A](la: List[A]): ListOps[A] = new ListOps(la)
 }
 
 final class ListOps[A](val la: List[A]) extends AnyVal {
+
+  /**
+    * Returns an Option of NonEmptyList from a List
+    *
+    * Example:
+    * {{{
+    * scala> import cats.data.NonEmptyList
+    * scala> import cats.implicits._
+    *
+    * scala> val result1: List[Int] = List(1, 2)
+    * scala> result1.toNel
+    * res0: Option[NonEmptyList[Int]] = Some(NonEmptyList(1, 2))
+    *
+    * scala> val result2: List[Int] = List.empty[Int]
+    * scala> result2.toNel
+    * res1: Option[NonEmptyList[Int]] = None
+    * }}}
+    */
   def toNel: Option[NonEmptyList[A]] = NonEmptyList.fromList(la)
+
+  /**
+    * Returns a IorNel from a List
+    *
+    * Example:
+    * {{{
+    * scala> import cats.data.IorNel
+    * scala> import cats.implicits._
+    *
+    * scala> val result1: List[String] = List("error 1", "error 2")
+    * scala> result1.toIorNel(1)
+    * res0: IorNel[String, Int] = Left(NonEmptyList(error 1, error 2))
+    *
+    * scala> val result2: List[String] = List.empty[String]
+    * scala> result2.toIorNel(1)
+    * res1: IorNel[String, Int] = Right(1)
+    * }}}
+    */
+  def toIorNel[B](ifEmpty: => B): IorNel[A, B] =
```

Comment:
The intention of this isn't very clear. I mean if you read a code `result1.toIorNel(1)`, it's not obvious that the `result1` is the error list while `1` is the valid result. Users will have to read the code here to understand the intention.  I am not sure about such syntax addition.